contribs/ev/src/main/java/org/matsim/contrib/ev/routing/EVNetworkRoutingModule.java

```diff
@@ -207,11 +207,16 @@ public EVNetworkRoutingModule(final String mode, final Network network, RoutingM
 
 			double consumption = driveEnergyConsumption.calcEnergyConsumption(l, travelT, Time.getUndefinedTime())
 					+ auxEnergyConsumption.calcEnergyConsumption(basicLeg.getDepartureTime(), travelT, l.getId());
+			if (consumption > 0) {
+				pseudoVehicle.getBattery().discharge(consumption);
+			} else {
+				pseudoVehicle.getBattery().discharge(-consumption);
```

Comment:
@jfbischoff,  did you mean this?
`pseudoVehicle.getBattery().charge(-consumption);`

Reply:
whoop, indeed.